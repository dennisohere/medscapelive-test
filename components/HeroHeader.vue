<template>
  <div class="hero-header_wrapper">
    <v-img
        class="hero-header_bg-image"
        data-test="hero-header__hero-img"
        cover
        :src="heroImage.src"
        :srcset="heroImage.srcSet"
        :title="heroImage.title"
        :alt="heroImage.alt"
    />

    <div class="top" data-test="hero-header__header">
      <router-link :to="HOME">
        <img
            :src="$uiService.image.medscapeLiveLogo.src"
            :alt="$uiService.image.medscapeLiveLogo.alt"
            :title="$uiService.image.medscapeLiveLogo.title"
        />
      </router-link>

      <slot :name="HERO_HEADER_RIGHT_ITEM_SLOT" />
    </div>

    <div class="content">
      <slot :name="HERO_HEADER_CONTENT_SLOT" />
    </div>
  </div>
</template>
<script lang="ts">
import { HOME } from '~/core/constants/router-paths'
import { MEDSCAPE_LIVE_LOGO_WHITE } from '@/core/constants/asset-paths'
import { type ImgAttribute } from '@/models/view/ImgAttribute'
import {
  HERO_HEADER_CONTENT_SLOT,
  HERO_HEADER_RIGHT_ITEM_SLOT
} from '~/core/constants/slot-names'

export default {
  name: 'HeroHeader',
  props: {
    heroImage: {
      type: Object as () => ImgAttribute,
      required: true
    }
  },
  data() {
    return {
      HOME,
      HERO_HEADER_CONTENT_SLOT,
      HERO_HEADER_RIGHT_ITEM_SLOT,
      MEDSCAPE_LIVE_LOGO_WHITE
    }
  }
}
</script>
<style lang="scss">
@import '@/styles/variables.scss';
.hero-header_wrapper {
  width: 100%;
  height: 100%;
  min-height: 200px;
  overflow: hidden;
  position: relative;

  .hero-header_bg-image {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center;
    z-index: 1;
  }

  .top {
    position: relative;
    z-index: 2;
    width: 100%;
    display: flex;
    justify-content: space-between;
    padding: 24px;

    img {
      width: auto;
      height: 30px;
    }
  }

  .content {
    position: relative;
    z-index: 2;
    display: flex;
    justify-content: center;
    padding: 0 24px;
  }
}
</style>
