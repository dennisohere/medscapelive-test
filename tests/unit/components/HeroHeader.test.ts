import { describe, it, expect, beforeEach, vi } from 'vitest'
import {mount, RouterLinkStub, type VueWrapper} from '@vue/test-utils'
import HeroHeader from '@/components/HeroHeader.vue'
import {createNuxtTestWrapper} from "~/tests/unit/setup/create-wrapper";

vi.mock('~/core/constants/slot-names', () => ({
  HERO_HEADER_CONTENT_SLOT: 'hero-header-content',
  HERO_HEADER_RIGHT_ITEM_SLOT: 'hero-header-right-item'
}))


describe('HeroHeader.vue', () => {
  let wrapper: VueWrapper
  beforeEach(async () => {
    wrapper = createNuxtTestWrapper(HeroHeader, {
      props: {
        heroImage: {
          src: 'yomommas-img-src',
          srcSet: 'yommommas-src-set',
          title: 'some title',
          alt: 'some alt'
        }
      },
    })
  })
  describe('render', () => {
    it('matches the snapshot', () => {
      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
