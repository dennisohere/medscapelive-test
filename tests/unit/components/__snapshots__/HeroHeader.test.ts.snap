// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroHeader.vue > render > matches the snapshot 1`] = `
<div
  class="hero-header_wrapper"
>
  <v-img-stub
    absolute="false"
    alt="some alt"
    class="hero-header_bg-image"
    cover="true"
    data-test="hero-header__hero-img"
    eager="false"
    inline="false"
    options="[object Object]"
    src="yomommas-img-src"
    srcset="yommommas-src-set"
    tile="false"
    title="some title"
    transition="fade-transition"
  />
  <div
    class="top"
    data-test="hero-header__header"
  >
    <router-link
      to="/"
    >
      <img
        alt="white logo of medscape live events"
        src="/static_vue/medscape-live-web/img/medscape-live-white.png"
        title="white medscape live logo"
      />
    </router-link>
    
    
  </div>
  <div
    class="content"
  >
    
    
  </div>
</div>
`;
